NetProbe/Model/NetworkMeasurement.mm

```diff
@@ -14,43 +14,41 @@
 static void setup_idempotent() {
     static bool initialized = false;
     if (!initialized) {
-
         // Set the logger verbose and make sure it logs on the "logcat"
-        mk::set_verbose(1);
+        mk::set_verbosity(MK_LOG_DEBUG2);
         // XXX Ok to call NSLog() from another thread?
-        mk::on_log([](const char *s) {
+        mk::on_log([](uint32_t, const char *s) {
             NSLog(@"%s", s);
         });
-
-        // Copy DNS resolver(s) from device
-        // This code copied from the iOS tutorial
-        // XXX This should run before every test but to do this we need
-        // to further change measurement-kit DNS code
-        res_state res = nullptr;
-        do {
-            res = (res_state) malloc(sizeof(struct __res_state));
-            if (res == nullptr) break;
-            if (res_ninit(res) != 0) break;
-            mk::clear_nameservers();
-            for (int i = 0; i < res->nscount; ++i) {
-                char addr[INET_ADDRSTRLEN];
-                if (inet_ntop(AF_INET, &res->nsaddr_list[i].sin_addr, addr,
-                            sizeof (addr)) == nullptr) {
-                    continue;
-                }
-                mk::add_nameserver(addr);
-                NSLog(@"adding DNS resolver: %s", addr);
-            }
-            free(res);
-            res = nullptr;
-        } while (0);
-        if (res) free(res);
-
-        // Remember that we have initialized
         initialized = true;
     }
 }
 
+static std::string get_dns_server() {
+    std::string dns_server = "8.8.4.4:53";
+    res_state res = nullptr;
+    res = (res_state) malloc(sizeof(struct __res_state));
+    if (res == nullptr) {
+        return dns_server;
+    }
+    if (res_ninit(res) != 0) {
+        free(res);
+        return dns_server;
+    }
+    for (int i = 0; i < res->nscount; ++i) {
+        char addr[INET_ADDRSTRLEN];
+        if (inet_ntop(AF_INET, &res->nsaddr_list[i].sin_addr, addr,
+                      sizeof (addr)) == nullptr) {
+            continue;
+        }
+        NSLog(@"found DNS resolver: %s", addr);
+        dns_server = addr;
+        break;
+    }
+    free(res);
+    return dns_server;
+}
+
```

Comment:
This is a rewrite of the algorithm above. Please check I made no stupid mistakes here :-).